File: src/types/notification.ts
// 个人业务通知（铃铛）DTO。通知只是入口提示，绝不作为授权依据；
// 后端读取时仍会按当前身份重新校验目标可见性。

export interface NotificationTargetDTO {
  route_key: string;
  resource_id: string;
}

export interface BusinessNotificationDTO {
  id: string;
  event_type: string;
  category: string;
  title: string;
  summary: string;
  created_at: string;
  is_read: boolean;
  read_at: string | null;
  project_name: string | null;
  object_name: string;
  task_status: "needs_action" | "submitted" | "processing" | "completed" | "partial" | "failed";
  task_group: "my_tasks" | "running_jobs" | "attention_items" | "recent_completed";
  action_required: boolean;
  next_action_label: string;
  delivery_status: string;
  target: NotificationTargetDTO;
}

export interface BusinessNotificationListResponseDTO {
  items: BusinessNotificationDTO[];
  total: number;
  page: number;
  page_size: number;
  unread_count?: number;
  categories?: string[];
}

export interface UnreadCountResponseDTO {
  unread_count: number;
}

export interface MarkReadBatchResponseDTO {
  requested_count: number;
  marked_count: number;
  already_read_count: number;
}

File: backend/app/schemas/notification.py
"""Safe first-party business notification API schemas."""

from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class NotificationTarget(BaseModel):
    route_key: str
    resource_id: uuid.UUID


class BusinessNotificationOut(BaseModel):
    id: uuid.UUID
    event_type: str
    category: str
    title: str
    summary: str
    created_at: datetime
    is_read: bool
    read_at: datetime | None
    project_name: str | None = None
    object_name: str
    task_status: str
    task_group: str
    action_required: bool
    next_action_label: str
    failure_reason: str | None = None
    recovery_suggestion: str | None = None
    target: NotificationTarget


class BusinessNotificationListResponse(BaseModel):
    items: list[BusinessNotificationOut]
    total: int
    page: int
    page_size: int
    unread_count: int
    pending_count: int
    categories: list[str] = Field(default_factory=list)


class UnreadCountResponse(BaseModel):
    unread_count: int


class MarkReadBatchRequest(BaseModel):
    notification_ids: list[uuid.UUID] = Field(min_length=1, max_length=100)


class MarkReadBatchResponse(BaseModel):
    requested_count: int
    marked_count: int
    already_read_count: int
